flow/activities/flowable_core.go

```diff
@@ -93,6 +151,57 @@ func (a *FlowableActivity) applySchemaDeltas(
 	return nil
 }
 
+// this is the updated approach of applying schema deltas to catalog. Unlike v1,
+// we use `table_schema_mapping` from catalog as base, and add new columns from
+// schemaDeltas that are not in the existing mapping. This function returns
+// a copy of schemasInCatalog with schemaDeltas applied.
+func applySchemaDeltaV2(
+	ctx context.Context,
+	schemasInCatalog map[string]*protos.TableSchema,
+	schemaDeltas []*protos.TableSchemaDelta,
+) (map[string]*protos.TableSchema, error) {
+	logger := internal.LoggerFromCtx(ctx)
+
+	schemasCopy := make(map[string]*protos.TableSchema, len(schemasInCatalog))
+	for tableName, schema := range schemasInCatalog {
+		if schema == nil {
+			return nil, fmt.Errorf("failed to deep copy table schema from catalog: table %s has nil schema", tableName)
+		}
+		schemasCopy[tableName] = proto.Clone(schema).(*protos.TableSchema)
+	}
+
+	ddlTableNames := make([]string, 0, len(schemaDeltas))
+	for _, schemaDelta := range schemaDeltas {
+		ddlTableNames = append(ddlTableNames, schemaDelta.DstTableName)
+	}
+
+	if len(schemasCopy) != len(ddlTableNames) {
+		matchedTables := make([]string, 0, len(schemasCopy))
+		for name := range schemasCopy {
+			matchedTables = append(matchedTables, name)
+		}
+		logger.Warn(fmt.Sprintf("not all tables are found, expected: %v, actual, %v", ddlTableNames, matchedTables))
```

Comment:
```suggestion
		logger.Warn(fmt.Sprintf("not all tables are found, expected: %v, actual: %v", ddlTableNames, matchedTables))
```
but really use slog.Any